packages/agent/src/lib/agents/codex.ts

```diff
@@ -62,32 +62,55 @@ export class CodexAgent extends BaseAgent {
   ): Promise<void> {
     const args = ['mcp', 'add'];
 
-    if (transport !== 'stdio') {
-      throw new Error(`${this.name} only supports stdio transport`);
-    }
+    if (transport === 'stdio') {
+      if (headers.length > 0) {
+        console.log(
+          colors.yellow(` ${this.name} does not support headers in stdio mode.`)
+        );
+      }
+
+      envs.forEach(env => {
+        if (/\w+=\w+/.test(env)) {
+          args.push(`--env=${env}`);
+        } else {
+          console.log(
+            colors.yellow(
+              ` Invalid ${env} environment variable. Use KEY=VALUE format`
+            )
+          );
+        }
+      });
 
-    envs.forEach(env => {
-      if (/\w+=\w+/.test(env)) {
-        args.push(`--env=${env}`);
-      } else {
+      args.push(name, '--', ...commandOrUrl.split(' '));
+    } else if (transport === 'http') {
+      if (envs.length > 0) {
         console.log(
           colors.yellow(
-            ` Invalid ${env} environment variable. Use KEY=VALUE format`
+            ` ${this.name} only supports environment variables in stdio mode.`
           )
         );
       }
-    });
 
-    if (headers.length > 0) {
-      console.log(
-        colors.yellow(
-          ` ${this.name} does not support HTTP or SSE servers. Ignoring headers.`
-        )
+      // TODO: allow the user to specify what envvar contains the
+      // bearer token.
+      //
+      // Codex works a bit different here: does not support
+      // arbitrary headers at this time but --bearer-token-env-var,
+      // the name of an envvar that holds the bearer token..
+
+      if (headers.length > 0) {
+        console.log(
+          colors.yellow(` ${this.name} does not support arbitrary headers.`)
+        );
+      }
```

Comment:
Ohh, interesting. Okey, for now let's push the fix and we will work on enabling this in a separate PR 👍